### ISSUE TITLE
Support PubSub payloads that are not JSON serializable

### ISSUE THREAD

**Issue by danielmcmillan** [aws-amplify/amplify-js#11135]
### Is this related to a new or existing API?

PubSub

### Describe the feature you'd like to request

While MQTT supports arbitrary binary data, there seems to be an undocumented limitation in the library that it can only handle JSON serializable messages.

Supporting arbitrary binary message payloads would enable communication with IoT devices that don't use JSON serialization.

### Describe the solution you'd like

The best solution may be to explicitly specify the message format to avoid any compatibility issue. If binary is specified, then the message for `publish` and `subscribe` would be an `ArrayBuffer` or view. Otherwise, existing behaviour is maintained.

### Describe alternatives you've considered

Another library such as "aws-iot-device-sdk-v2"

### Is this something that you'd be interested in working on?

- [X] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by cwomack** [comment]
Hello, @danielmcmillan and thanks for opening up this issue.  I've created [#5296](https://github.com/aws-amplify/docs/issues/5296) within the docs repo to improve the documentation regarding this current limitation of PubSub as you've described.  We've also added this to the list for feature requests within PubSub to add support for non-JSON serializable payloads. 

We'll update this issue as we make progress on the feature request side, but appreciate you calling out the need for better documentation for PubSub as well!
